### PR TITLE
[docs] Add `PickersPopper` component to customization playground

### LINKED ISSUE #15263: [docs] Date Picker customization `borderRadius` does not round all components
### Steps to reproduce

Link to live example: https://mui.com/x/react-date-pickers/date-picker/#customization

Steps:
1. select "DesktopDatePicker"
2. select DateCalendar from components dropdown
3. set borderRadius for maximum (20px)

### Current behavior

DateCalendar is rounded, but there is MuiPaper without rounding still visible (overriding MuiPaper with border-radius: "20px" gives the desired rounding)
image

### Expected behavior

DateCalendar is rounded correctly
image

### Context

DateCalendar rounding to be fixed, as it is useless to pass the ‘borderRadius’ property as the other element below it is not rounded.

### Your environment

---

**Search keywords**: DateCalendar borderRadius DesktopDatePicker

### PR DESCRIPTION
**Pull request by LukasTy**
Fixes #15263.

### CODE REVIEW

File: docs/data/date-pickers/date-picker/examplesConfig.styling.tsx [hunk 1]
```diff
@@ -290,6 +290,30 @@ export const datePickerExamples: PickersSubcomponentType = {
...
+        type: 'warning',
+        parentSlot: 'popper',
+        current: true,
+        comments:
+          'Because of the structure of the DesktopDatePicker, the `sx` prop needs to be applied to the `popper` slot',
+      },
+      styledComponents: {
+        type: 'warning',
+        parentSlot: 'desktopPaper',
+        parentComponent: 'Paper',
```
Comment (LukasTy) [inline docs/data/date-pickers/date-picker/examplesConfig.styling.tsx:309]: `PickersPopper` is an `internal` component, I think that it is better to show the compromise of overriding it with `MuiPaper` instead. 🤔

**Comment by flaviendelangle** [comment]
image

I did not check the changes in depth, but as is it's a weird experience to play with the `PickersPopper` on the customization playground :grimacing:

**Comment by LukasTy** [comment]
Are you suggesting removing the `root` slot to reduce the cases of "unclear" result states? 🤔

**Comment by flaviendelangle** [comment]
Do you see some customization use cases for the `root`?
Maybe we could at least change the dedault selected slot on the UI

**Comment by LukasTy** [comment]
> Do you see some customization use cases for the `root`?
> Maybe we could at least change the dedault selected slot on the UI

I have changed the order of the slots and the first one (paper) is now the default.

**Pull request merged**